packages/core/src/Layout/Layout.ts

```diff
@@ -1,14 +1,34 @@
-const { v4: uuidv4 } = require('uuid');
+import { v4 as uuidv4 } from 'uuid';
+import { PageSources } from '../Page/PageSources';
+import { NodeProcessor } from '../html/NodeProcessor';
+import * as logger from '../utils/logger';
+import { ExternalManager, ExternalManagerConfig } from '../External/ExternalManager';
 
-const { PageSources } = require('../Page/PageSources');
-const { NodeProcessor } = require('../html/NodeProcessor');
+const LAYOUT_PAGE_BODY_VARIABLE = 'content';
 
-const logger = require('../utils/logger');
+export type LayoutConfig = ExternalManagerConfig & { externalManager: ExternalManager };
```

Comment:
I was thinking it could be better to have a type folder/file to consolidate types. What do you guys think? @yucheng11122017 @EltonGohJH @kaixin-hc

Reply:
I think for now I am okay with what we are doing right now.
I checked our repository for `export type` and I realised we dun really have much types and references in the first place. 

I am leaning more towards against this change. (due to the fact that we do not have much types and we do not often nid to reference the typing.) 

I feel that even if we want to do this, we should create a separate and do the 1 time migration PR rather than doing it first on this PR so that our repository is more coherent.

What do @yucheng11122017 @kaixin-hc think about this?

Reply:
I agree it should not be part of this PR - we can open an issue to discuss if its needed

Reply:
I agree with @EltonGohJH, resolving this conversation. I think I will open a new issue if the number of types used increases significantly but I think this is not an urgent issue for now.